mypyc/ir/rtypes.py

```diff
@@ -254,6 +254,16 @@ def __repr__(self) -> str:
 else:
     c_pyssize_t_rprimitive = int64_rprimitive
 
+# low level pointer, represented as integer in C backends
+int32_ptr_rprimitive = RPrimitive('int32 *', is_unboxed=True, is_refcounted=False,
+                              ctype='int32_t *', size=4)  # type: Final
+int64_ptr_rprimitive = RPrimitive('int64 *', is_unboxed=True, is_refcounted=False,
+                              ctype='int64_t *', size=8)  # type: Final
```

Comment:
I don't see the value of having the pointer target 32-bit integers on 32-bit platforms and 64-bit integers on 64-bit platforms. Since we also want to use integer operations on these (pointer arithmetic), the most reasonable choices for the type would be `char *` (for pointer arithmetic) or an unsigned integer type. 

I'm leaning towards using an unsigned integer type, so that we'll always have to explicitly cast to the desired pointer type in generated C. I think that this seems like a logical way to do this, as the pointers are untyped. Thus a plain pointer value should never be deferenced without a cast. (`void *` is not a good option since it doesn't support pointer arithmetic.)

So the pointer type could use `size_t` as the C type, similar to `CPyTagged`. However, `size_t` would look a bit strange in generated code, so we can add a typedef that resolves to it, for example, `CPyPtr`, defined in `mypyc_util.h` (similar to `CPyTagged`).